### PR TITLE
Create Common Pie Chart Component

### PR DESCRIPTION
**Pull request by harsh-jindal**
Fixes part of #2081 
Changes: Create Common Pie Chart Components. Created a common component for MyAnalytics and SkillUsageCard.

Demo Link: https://pr-2199-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

File: src/components/shared/PieChartContainer.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+  onPieEnter = (cell, index) => {
+    this.setState({
+      activePieIndex: index,
+    });
+  };
+
+  render() {
+    const { activePieIndex } = this.state;
+    const { cellData, data } = this.props;
```
Comment (akshatnitd) [inline src/components/shared/PieChartContainer.js]: Keep some extra props like `showLegend` since it is a shared component, so make it customizable.

Comment (akshatnitd) [inline src/components/shared/PieChartContainer.js]: Use - `const { cellData, data, ...otherProps } = this.props;`

File: src/components/shared/PieChartContainer.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            data={data}
+            cx={300}
+            cy={175}
+            innerRadius={80}
+            nameKey="deviceType"
+            dataKey="count"
+            outerRadius={120}
+            fill="#8884d8"
+            onMouseEnter={this.onPieEnter}
+          >
```
Comment (akshatnitd) [inline src/components/shared/PieChartContainer.js]: `{...otherProps}`

**Comment by harsh-jindal** [comment]
@akshatnitd Made the changes. Please review.

File: src/components/shared/PieChartContainer.js (hunk 1 above)
Comment (akshatnitd) [reply src/components/shared/PieChartContainer.js]: @harsh-jindal This comment isn't addressed in the PR.

Comment (harsh-jindal) [reply src/components/shared/PieChartContainer.js]: @akshatnitd I didn't get what you mean by showLegend ?

Comment (akshatnitd) [reply src/components/shared/PieChartContainer.js]: Keep a boolean prop for this component, `showLegend`, by which we can show/hide the Legend of the pie-chart.

Comment (akshatnitd) [reply src/components/shared/PieChartContainer.js]: @harsh-jindal Please finalise the PR

Comment (harsh-jindal) [reply src/components/shared/PieChartContainer.js]: @akshatnitd Done

**Comment by harsh-jindal** [comment]
@akshatnitd Made the changes. Please review

**Pull request merged**